### PR TITLE
Support mode: removed 24hrs check, now checking number of usb sticks …

### PR DESCRIPTION
**Pull request by andywerner**
…mounted

### CODE REVIEW

File: octoprint_mrbeam/support.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 _logger = mrb_logger("octoprint.plugins.mrbeam.support")
 
+USB_MOUNT_DIR = "/media/"
+
```
Comment (amrsoll) [inline octoprint_mrbeam/support.py:16]: Remove `SUPPORT_STICK_FILE_MAX_AGE`

File: octoprint_mrbeam/support.py [hunk 2]
```diff
@@ -21,13 +23,14 @@ def check_support_mode(plugin):
...
     support_mode_enabled = False
+    if not hasattr(plugin, "_settings") or not hasattr(plugin, "_user_manager"):
+        return None
+
     try:
         if plugin._settings.get(["dev", "support_mode"]) or (
             os.path.isfile(SUPPORT_STICK_FILE_PATH)
-            and time.time() - os.path.getmtime(SUPPORT_STICK_FILE_PATH)
-            < SUPPORT_STICK_FILE_MAX_AGE
+            and len(os.listdir(USB_MOUNT_DIR)) > 0  # if i a usb stick is plugged in
```
Comment (amrsoll) [inline octoprint_mrbeam/support.py]: Use [`os.path.isdir()`](https://docs.python.org/3/library/os.path.html#os.path.isdir)

File: octoprint_mrbeam/support.py [hunk 3]
```diff
@@ -46,13 +49,14 @@ def check_calibration_tool_mode(plugin):
...
     mode_enabled = False
+    if not hasattr(plugin, "_settings") or not hasattr(plugin, "_user_manager"):
+        return None
+
     try:
         if plugin._settings.get(["dev", "calibration_tool_mode"]) or (
             os.path.isfile(CALIBRATION_STICK_FILE_PATH)
-            and time.time() - os.path.getmtime(CALIBRATION_STICK_FILE_PATH)
-            < SUPPORT_STICK_FILE_MAX_AGE
+            and len(os.listdir(USB_MOUNT_DIR)) > 0  # if i a usb stick is plugged in
```
Comment (amrsoll) [inline octoprint_mrbeam/support.py]: Use [`os.path.isdir()`](https://docs.python.org/3/library/os.path.html#os.path.isdir)

**Review by amrsoll — CHANGES_REQUESTED** [review]

File: octoprint_mrbeam/support.py (hunk 1 above)
Comment (amrsoll) [reply octoprint_mrbeam/support.py:16]: well it didn't show in the refered lines here, but you should delete the definition

File: octoprint_mrbeam/support.py (hunk 2 above)
Comment (andywerner) [reply octoprint_mrbeam/support.py]: Nope. I want the dirs in there. The number of dirs must be 1 or more!

File: octoprint_mrbeam/support.py (hunk 3 above)
Comment (andywerner) [reply octoprint_mrbeam/support.py]: same

File: octoprint_mrbeam/support.py (hunk 2 above)
Comment (andywerner) [reply octoprint_mrbeam/support.py]: No. I want to test if the number of subdirs in there is at least 1.

**Review by amrsoll — APPROVED** [review]

**Pull request merged**